### PR TITLE
docs: Add deque to the table of content and fix ROI underline

### PR DESCRIPTION
**Pull request by tytan652**
### Description

Fixes warnings that sphinx reported that comes from "recent" changes:
- Add deque to libobs/util table of content
- Fix title underline of obs_encoder_roi

Also fix a deprecation note for circlebuf that was set to a placeholder version (`3X.0`).

### Motivation and Context

Fix docs warnings right after I found them.

### How Has This Been Tested?

`make html` and Sphinx didn't complain of the issues fixed by the changes and deque is shown in the libobs/util TOC.

### Types of changes

- Documentation (a change to documentation pages)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: docs/sphinx/reference-libobs-util.rst [hunk 1]
```diff
@@ -8,6 +8,7 @@ Platform/Utility API Reference (libobs/util)
    reference-libobs-util-bmem
    reference-libobs-util-circlebuf
```
Comment (derrod) [inline docs/sphinx/reference-libobs-util.rst:9]: Probably makes sense to remove this as well then.

**Review by derrod — APPROVED** [review]

File: docs/sphinx/reference-libobs-util.rst (hunk 1 above)
Comment (tytan652) [reply docs/sphinx/reference-libobs-util.rst:9]: If it's still in available in the API, it doesn't make sense to remove it even if it's "deprecated".
But the whole page should been updated to have all the function shown as deprecated, which is out of the scope that I want to keep for this PR.

Comment (RytoEX) [reply docs/sphinx/reference-libobs-util.rst:9]: @derrod Seems resolved?

File: docs/sphinx/reference-libobs-util.rst [hunk 2]
```diff
@@ -8,6 +8,7 @@ Platform/Utility API Reference (libobs/util)
...
    reference-libobs-util-config-file
+   reference-libobs-util-deque
    reference-libobs-util-darray
```
Comment (RytoEX) [inline docs/sphinx/reference-libobs-util.rst:12]:
```suggestion
   reference-libobs-util-darray
   reference-libobs-util-deque
```
Alphabetization.

**Review by RytoEX — CHANGES_REQUESTED** [review]

File: docs/sphinx/reference-libobs-util.rst (hunk 2 above)
Comment (tytan652) [reply docs/sphinx/reference-libobs-util.rst:12]: No because:
- Double-ended queue
- Dynamic Array
- Dynamic String

Comment (RytoEX) [reply docs/sphinx/reference-libobs-util.rst:12]: Are these _really_ alphabetized by their display name and not the name of the page?

Comment (tytan652) [reply docs/sphinx/reference-libobs-util.rst]: Realized that indeed those are alphabetized based on their identifiers and not the documentation name.

Comment (RytoEX) [reply docs/sphinx/reference-libobs-util.rst]: If the HTML rendered order ends up causing confusion, blame me, and we'll figure out something.

**Review by RytoEX — APPROVED** [review]

**Pull request merged**